src/Storefront/Resources/app/storefront/build/start-hot-reload.js

```diff
@@ -83,10 +83,10 @@ const proxyOptions = {
                     return;
                 }
                 // we only replace things when the request is a document
-                const isDocumentRequest = req.headers['sec-fetch-dest'] === 'document' || req.headers.accept.indexOf('text/html') !== -1;
-                const isHtmlRequests = req.url.indexOf('widgets/menu/offcanvas') !== -1 || req.url.indexOf('checkout/offcanvas') !== -1;
+                const isDocumentRequest = req.headers['sec-fetch-dest'] === 'document' || req.headers.accept?.indexOf('text/html') !== -1;
+                const isOffcanvasRequest = ['/widgets/menu/offcanvas', '/checkout/offcanvas'].some(path => req.url.includes(path));
```

Comment:
what about ajax modals? Would they be also affected? "we only replace things when the request is a document"

Reply:
they return also just html, so should be fine

Reply:
i would not check for a `sec-fetch-dest` header, it's chromium specific I guess